src/context-selection/data-set-selector-bar-item/data-set-selector-bar-item.js

```diff
@@ -0,0 +1,49 @@
+import i18n from '@dhis2/d2-i18n'
+import { SelectorBarItem } from '@dhis2/ui'
+import React, { useState } from 'react'
+import { MenuSelect } from '../menu-select/index.js'
+import { useDataSetId } from '../use-context-selection.js'
+import useDataSet from './use-data-set.js'
+import useSelectableDataSets from './use-selectable-data-sets.js'
+
+export default function DataSetSelectorBarItem() {
+    const [dataSetOpen, setDataSetOpen] = useState(false)
+    const [dataSetId, setDataSetId] = useDataSetId()
+    const { loadingDataSet, errorDataSet, dataSet } = useDataSet()
+
+    const {
+        loadingSelectableDataSets,
+        errorSelectableDataSets,
+        selectableDataSets,
+    } = useSelectableDataSets()
+
+    const selectorBarItemValue = loadingDataSet
+        ? i18n.t('Fetching data set info')
+        : errorDataSet
+        ? i18n.t('Error occurred while loading data set info')
+        : dataSet?.displayName
+
+    return (
+        <SelectorBarItem
+            label={i18n.t('Data set')}
+            value={dataSetId ? selectorBarItemValue : undefined}
+            open={dataSetOpen}
+            setOpen={setDataSetOpen}
+            noValueMessage={i18n.t('Choose a data set')}
+        >
+            {loadingSelectableDataSets && i18n.t('Fetching data sets')}
+            {errorSelectableDataSets &&
+                i18n.t('Error occurred while loading data sets')}
+            {selectableDataSets && (
+                <MenuSelect
+                    values={selectableDataSets}
+                    selected={dataSetId}
+                    onChange={({ selected }) => {
+                        setDataSetId(selected)
+                        setDataSetOpen(false)
+                    }}
+                />
+            )}
```

Comment:
I can imagine you could get some funky output here with stale while revalidate. Previous values won't always be cleared, so it might be possible that you have an error and stale data for example. To prevent that from happening you'd want separate returns for loading, error and data, in that order.

Reply:
I'll talk to @cooper-joe about it

Reply:
@cooper-joe and I talked about this and agreed on the following:

>  for now it's best to simply not show any data while loading new data / an error occurred. We can still think about how to improve in in the future (either generally or specifically for this app)

Reply:
Ok, so then this should be changed right? Because currently it will show data whilst revalidating/loading

Reply:
Ah I see you've addressed that with renderMenu. All right